src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -1,74 +1,100 @@
 """Collection of helper functions related to configuration of datasets dependent on backend."""
 
 import sys
-from typing import Dict, Union
+from typing import Union
 
 from hdmf.common import Data
 from pynwb import NWBFile, TimeSeries
 
+from ._configuration_models._base_backend import BackendConfiguration
 from ._configuration_models._hdf5_backend import HDF5BackendConfiguration
 from ._configuration_models._zarr_backend import ZarrBackendConfiguration
 from ..importing import is_package_installed
 
 
-def _resolve_nwbfile_object_ids(
+def _remap_backend_configuration_to_nwbfile(
```

Comment:
I do like this approach more, but what would you think about making it an instance method on the `backend_configuration`? Like `.equivocate_object_ids(self, nwbfile: pynwb.NWBFile)`

Reply:
We could also unfreeze the object ID field and adjust the `self: BackendConfiguration` in-place - WDYT?

Reply:
I was also thinking about those two but did not want to rock or boat or delay your other PR.

I think that both make sense!

Reply:
Sounds good, did you want to do that in this PR?

Reply:
Yeah, sure.

Reply:
Done.